### PR TITLE
:running: add tests for simplified webhook interface

### PR DESCRIPTION
**Pull request by mengqiy**
Followup PR per https://github.com/kubernetes-sigs/controller-runtime/pull/328#issuecomment-471138778

### CODE REVIEW

File: pkg/builder/build_test.go [hunk 1]
```diff
@@ -121,6 +120,93 @@ var _ = Describe("application", func() {
...
+				Owns(&appsv1.ReplicaSet{}).
+				Build(noop)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(instance).NotTo(BeNil())
+			svr := m.GetWebhookServer()
+			Expect(svr).NotTo(BeNil())
+
+			By("trying to register an existing mutating webhook path")
+			path := generateMutatePath(schema.GroupVersionKind{Group: "foo.test.org", Version: "v1", Kind: "TestDefaulter"})
+			Ω(func() { svr.Register(path, nil) }).Should(Panic())
```
Comment (DirectXMan12) [inline pkg/builder/build_test.go]: this is a bit of a weak test -- what if we add validation that handler is not nil, or something?  Maybe we should check that it responds to a valid admission request?

File: pkg/builder/builder_suite_test.go [hunk 2]
```diff
@@ -42,7 +44,88 @@ var cfg *rest.Config
 var _ = BeforeSuite(func(done Done) {
 	logf.SetLogger(zap.LoggerTo(GinkgoWriter, true))
 
-	testenv = &envtest.Environment{}
+	testenv = &envtest.Environment{
+		CRDs: []*apiextensionsv1beta1.CustomResourceDefinition{
+			{
+				TypeMeta: metav1.TypeMeta{
```
Comment (DirectXMan12) [inline pkg/builder/builder_suite_test.go]: refactor this out into a method that takes a kind name

File: pkg/builder/build_test.go (hunk 1 above)
Comment (mengqiy) [reply pkg/builder/build_test.go]: Done.
PTAL

**Comment by DirectXMan12** [comment]
(as discussed) inject isn't quite it the right spot now (you can register hooks before adding to the manager)

**Comment by DirectXMan12** [comment]
/lgtm
/approve

We'll need to resolve the dep management issue at some point, but this is fine for now.

**Pull request merged**